qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -460,267 +860,566 @@ void IndirectFitAnalysisTab::updatePlot(
- * @param addParameters Adds parameters from function to QtProperty if True,
- *                      else returns empty QtProperty.
- */
-QtProperty *
-IndirectFitAnalysisTab::createFunctionProperty(QtProperty *functionGroup,
-                                               const bool &addParameters) {
-  const auto &functionName = functionGroup->propertyName();
-  const auto parameters = getFunctionParameters(functionName);
-
-  for (const auto &parameter : parameters) {
-    QString paramName = functionName + "." + parameter;
-    m_properties[paramName] = m_dblManager->addProperty(parameter);
-    m_dblManager->setDecimals(m_properties[paramName], NUM_DECIMALS);
-
-    if (addParameters)
-      functionGroup->addSubProperty(m_properties[paramName]);
-  }
-  return functionGroup;
+/**
+ * Ensures one workspace is able to be appended to another.
+ *
+ * @param inputWS   The workspace to maintain compatibility with.
+ * @param spectraWS The workspace to make compatible.
+ */
+void IndirectFitAnalysisTab::ensureAppendCompatibility(
+    MatrixWorkspace_sptr inputWS, MatrixWorkspace_sptr spectraWS) const {
+  spectraWS->setInstrument(inputWS->getInstrument());
+  spectraWS->replaceAxis(0, inputWS->getAxis(0)->clone(spectraWS.get()));
+  spectraWS->setDistribution(inputWS->isDistribution());
+}
+
+/**
+ * Creates an algorithm for extracting the spectra from a specified workspace.
+ *
+ * @param inputWS     The workspace to extract spectra from.
+ * @param startIndex  The index of the first spectrum to be retained.
+ * @param endIndex    The index of the last spectrum to be retained.
+ * @param startX      The x-value that is within the first bin to be retained.
+ * @param endX        The x-value that is within the last bin to be retained.
+ * @return            An algorithm for extracting spectra.
+ */
+IAlgorithm_sptr IndirectFitAnalysisTab::extractSpectraAlgorithm(
+    MatrixWorkspace_sptr inputWS, int startIndex, int endIndex, double startX,
+    double endX) const {
+  IAlgorithm_sptr extractSpectraAlg =
+      AlgorithmManager::Instance().create("ExtractSpectra");
+  extractSpectraAlg->initialize();
+  extractSpectraAlg->setChild(true);
+  extractSpectraAlg->setLogging(false);
+  extractSpectraAlg->setProperty("InputWorkspace", inputWS);
+  extractSpectraAlg->setProperty("StartWorkspaceIndex", startIndex);
+  extractSpectraAlg->setProperty("XMin", startX);
+  extractSpectraAlg->setProperty("XMax", endX);
+  extractSpectraAlg->setProperty("EndWorkspaceIndex", endIndex);
+  extractSpectraAlg->setProperty("OutputWorkspace", "__extracted");
+  return extractSpectraAlg;
+}
+
+/**
+ * Creates an algorithm for appending the spectra of a specified workspace to
+ * another specified workspace.
+ *
+ * @param inputWS   The workspace to append to.
+ * @param spectraWS The workspace containing the spectra to append.
+ * @return          An algorithm for appending spectra.
+ */
+IAlgorithm_sptr IndirectFitAnalysisTab::appendSpectraAlgorithm(
```

Comment:
Could you just append the workspaces and return the result?
`ConvFit` which inherits `IndirectFitAnalysisTab` has method `appendAlgorithm` doing almost the same job as this one.